subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -1194,11 +1194,10 @@ uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
 
 	/* Get Adv data from function parameters */
 	if (hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_AD_DATA) {
-		ad_data = hdr_data;
-		ad_len = *ad_data;
-		++ad_data;
+		ad_len = *(uint8_t *)hdr_data;
+		hdr_data = (uint8_t *)hdr_data + sizeof(ad_len);
 
-		ad_data = (void *)sys_get_le32(ad_data);
+		(void)memcpy(&ad_data, hdr_data, sizeof(ad_data));
```

Comment:
Since `ad_data` is already a pointer, shouldn't it be
```suggestion
		(void)memcpy(ad_data, hdr_data, sizeof(ad_data));
```

Reply:
No, here pointer data type is being assigned, not the pointed data. Use of memcpy allows portability to machine with differing address/pointer sizes.